packages/hardhat-core/test/internal/hardhat-network/helpers/useProvider.ts

```diff
@@ -115,6 +123,77 @@ export function useProvider({
         this.server.getProvider()
       );
     }
+
+    if (rethnetBinary !== undefined) {
```

Comment:
Is there a reason why you didn't put this code in a dedicated class that gets constructed there and can also be reused elsewhere?

Reply:
Only that I didn't foresee any specific other usage for such an abstraction. Usually I avoid preparing for unforeseen re-use as a premature optimization.

Reply:
Leaving aside other use cases, how would you ensure encapsulation of your code in that case? (https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4095/commits/50378a0b033586fe478d4ed35d91a5148566153f